### PR TITLE
fix(core:accordion): a11y fixes and improvements

### PR DESCRIPTION
**Pull request by jeeyun**
- We have a wrapper container that groups an accordion header and its content. We call this a 'panel' but this leads to a lot of confusion as the [aria- documentation refers to just the content part as 'panel'](https://www.w3.org/TR/wai-aria-practices-1.1/#accordion). We'll rename the container to avoid this confusion.
- The accordion content has `role='region'` but moving this up to the host element for better Screen Reader support
- Adding `aria-labelledby` and `aria-controls` attributes and ID values auto-assigned (can be overriden if they use their own ID values)

Signed-off-by: Jeeyun Lim <jeeyun.lim@gmail.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

We are renaming `cds-accordion-panel` to `cds-accordion-section`. Since accordion is still in beta, this is a good time to introduce this breaking change. The migration path would be to rename the tags in the applications appropriately.

### CODE REVIEW

**Comment by bbogdanov** [comment]
We should use `a11y` type in the commit message `a11y(core:accordion): fixes and improvements`.

File: apps/adoption/src/app/migrations/accordion/accordion.1.core.txt
Comment (bdryanovski) [inline apps/adoption/src/app/migrations/accordion/accordion.1.core.txt:7]: Thanks for addressing this ;)

**Review by bdryanovski — APPROVED** [review]
LGTM

File: apps/core-angular-cli/src/app/app.component.html
Comment (mathisscott) [inline apps/core-angular-cli/src/app/app.component.html:27]: Is this not a breaking change?

File: packages/core/src/accordion/accordion-content.element.ts
Comment (mathisscott) [inline packages/core/src/accordion/accordion-content.element.ts:49]: Does id need to be a property?

File: packages/core/src/accordion/accordion-header.element.ts
Comment (mathisscott) [inline packages/core/src/accordion/accordion-header.element.ts:48]: Same here. Should id be a lit property? Disregard if it already is. I'm reviewing on my phone and what I can see of the file is limited

**Review by mathisscott — CHANGES_REQUESTED** [review]
This seems like a breaking change to me. As in, it would break anyone who is currently using the accordion and updates. 

As such, I think we should wait until v6 to merge this in.

File: packages/core/src/accordion/accordion-content.element.ts
Comment (hippee-lee) [reply packages/core/src/accordion/accordion-content.element.ts:49]: I think id is inherited from LitElement but now I can't find confirmation of that from the docs.

Comment (coryrylan) [reply packages/core/src/accordion/accordion-content.element.ts:49]: `id` is an inherited property because `LitElement` extends `HTMLElement`. So all our components automatically get `id` without having to have `@property` . The nice part is if you set the `id` property it automatically reflects to the `id` attribute.

File: apps/core-angular-cli/src/app/app.component.html
Comment (coryrylan) [reply apps/core-angular-cli/src/app/app.component.html:27]: I think this is a breaking change. Accordion with a few others were moved out of beta in march. This would be good for 6.0

File: packages/core/src/accordion/accordion-content.element.ts
Comment (mathisscott) [reply packages/core/src/accordion/accordion-content.element.ts:49]: I agree that `id` is part of the DOM spec for HTMLElement. My question was more concerned about whether we want this to be Lit property because of the tie-in to the update lifecycle and the React wrapper. React already has problems with us updating the DOM without it knowing. So I tend to only do this (updating the DOM outside of the Lit lifecycle) for stealth attributes that drive CSS selectors or internal logic (like with Clarity Motion). Things React wouldn't need to keep track of in the Virtual DOM.

Comment (jeeyun) [reply packages/core/src/accordion/accordion-content.element.ts:49]: I also updated the tree-view PR to not be a Lit property per @coryrylan 's suggestion. Either way we pick, we should keep it consistent across components which way we go.

File: apps/core-angular-cli/src/app/app.component.html
Comment (jeeyun) [reply apps/core-angular-cli/src/app/app.component.html:27]: Yes, this is a breaking change. Makes sense to go into 6.0. 👍

File: packages/core/src/accordion/accordion-content.element.ts
Comment (coryrylan) [reply packages/core/src/accordion/accordion-content.element.ts:49]: Yeah all good points @mathisscott  So many DOM props/attr are not consistent in staying in sync together.

File: packages/core/src/accordion/accordion-section.element.ts
Comment (scroniser) [inline packages/core/src/accordion/accordion-section.element.ts:84]: @gnomeontherun - aria-expanded should be associated with the button element, as it was previously.

**Comment by hippee-lee** [comment]
I am closing this in favor of two PR's that will replace it. 
[This](https://github.com/vmware/clarity/pull/6200) PR contains the accessibility changes that can be merged as soon as they are approved. [This](https://github.com/vmware/clarity/pull/6195) PR contains the renamed element changes that are a breaking change slated for v6.

Thank you @jeeyun for this and all of your contributions!!! ♥️

**Pull request closed without merging**
